File: sqlparser/src/sql/Dialect.java
//@
package cdata.sql;
//@
/*#
using RSSBus.core;
using RSSBus;
namespace CData.Sql {
#*/

import core.ParserCore;

public abstract class Dialect {
  public /*#virtual#*/ SqlStatement parse(SqlTokenizer tokenizer) throws Exception {
    return null;
  }

  public /*#virtual#*/ SqlExpression readTerm(SqlTokenizer tokenizer) throws Exception {
    return null;
  }

  public /*#virtual#*/ String build(SqlStatement stmt) throws Exception {
    return null;
  }

  public /*#virtual#*/ String writeTerm(SqlExpression column) throws Exception {
    return null;
  }

  public /*#virtual#*/ String getCustomCompareOp(SqlTokenizer tokenizer) throws Exception {
    return null;
  }

  public /*#virtual#*/ String buildTableAlias(SqlTable table) throws Exception {
    return null;
  }

  public /*#virtual#*/ String parseIdentifierName(SqlToken token) throws Exception {
    return null;
  }

  public /*#virtual#*/ String buildLimitOffset(SqlExpression limit, SqlExpression offset) throws Exception {
    return null;
  }

  public /*#virtual#*/ AlterAction parseAlterAction(SqlTokenizer tokenizer) throws Exception {
    return null;
  }

  public /*#virtual#*/ String buildAlterAction(AlterAction tableAction) throws Exception {
    return null;
  }

  public /*#virtual#*/ SqlColumnDefinition parseColumnDefinition(SqlTokenizer tokenizer) throws Exception {
    return null;
  }

  public /*#virtual#*/ String buildColumnDefinition(SqlColumnDefinition definition) throws Exception {
    return null;
  }

  public /*#virtual#*/ String encodeIdentifier(String identifier, String openQuote, String closeQuote) throws Exception {
    return null;
  }

  public /*#virtual#*/ String buildTableConstraint(SqlCollection<SqlColumnDefinition> definitions) throws Exception {
    return null;
  }

  public /*#virtual#*/ SqlExpression readOption(SqlTokenizer tokenizer) throws Exception {
    return null;
  }

  public /*#virtual#*/ String writeOption(SqlExpression option) throws Exception {
    return null;
  }

  public /*#virtual#*/ RebuildOptions getRebuildOptions() {
    return null;
  }

  public /*#virtual#*/ Character getEscapeChar() {
    return '\\';
  }

  public static SqlQueryStatement parseSelectUnion(SqlTokenizer tokenizer, Dialect dialect) throws Exception {
    return ParserCore.parseSelectUnion(tokenizer, dialect);
  }

  public static SqlExpression readExpression(SqlTokenizer tokenizer, SqlStatement stmt) throws Exception {
    return ParserCore.readExpression(tokenizer, stmt);
  }
}

File: sqlparser/src/sql/csgen/Dialect.cs
using System;
using System.Collections;
using System.IO;
using System.Threading;
using System.Net;
using System.Net.Sockets;
using System.Text;
using RSSBus.core.j2cs;



using RSSBus.core;
using RSSBus;
namespace CData.Sql {


public abstract class Dialect {
  public virtual SqlStatement Parse(SqlTokenizer tokenizer) {
    return null;
  }

  public virtual SqlExpression ReadTerm(SqlTokenizer tokenizer) {
    return null;
  }

  public virtual string Build(SqlStatement stmt) {
    return null;
  }

  public virtual string WriteTerm(SqlExpression column) {
    return null;
  }

  public virtual string GetCustomCompareOp(SqlTokenizer tokenizer) {
    return null;
  }

  public virtual string BuildTableAlias(SqlTable table) {
    return null;
  }

  public virtual string ParseIdentifierName(SqlToken token) {
    return null;
  }

  public virtual string BuildLimitOffset(SqlExpression limit, SqlExpression offset) {
    return null;
  }

  public virtual AlterAction ParseAlterAction(SqlTokenizer tokenizer) {
    return null;
  }

  public virtual string BuildAlterAction(AlterAction tableAction) {
    return null;
  }

  public virtual SqlColumnDefinition ParseColumnDefinition(SqlTokenizer tokenizer) {
    return null;
  }

  public virtual string BuildColumnDefinition(SqlColumnDefinition definition) {
    return null;
  }

  public virtual string EncodeIdentifier(string identifier, string openQuote, string closeQuote) {
    return null;
  }

  public virtual string BuildTableConstraint(SqlCollection<SqlColumnDefinition> definitions) {
    return null;
  }

  public virtual SqlExpression ReadOption(SqlTokenizer tokenizer) {
    return null;
  }

  public virtual string WriteOption(SqlExpression option) {
    return null;
  }

  public virtual RebuildOptions GetRebuildOptions() {
    return null;
  }

  public virtual char GetEscapeChar() {
    return '\\';
  }

  public static SqlQueryStatement ParseSelectUnion(SqlTokenizer tokenizer, Dialect dialect) {
    return ParserCore.ParseSelectUnion(tokenizer, dialect);
  }

  public static SqlExpression ReadExpression(SqlTokenizer tokenizer, SqlStatement stmt) {
    return ParserCore.ReadExpression(tokenizer, stmt);
  }
}
}

